### ISSUE TITLE
Path must be connected

### ISSUE THREAD

**Issue by saraedum** [flatsurf/flatsurf#137]
some examples fail with the "Path must be connected" assertion. One such example is
```
FlatTriangulationCombinatorial(vertices = (1, -3, 25, 22, 4, -6, -21, -23, -25, -26, -27, 10, -12, 8, -7, 12, -11, 13, -15, 26, 3, -2)(-1, 5, -4, 20, 18, 9, -8, 7, -9, 17, 14, -13, -16, -17, -18, -19, 6, -5, 2)(-10, 24, 21, 19, -20, -22, 23, -24, 27, 15, -14, 16, 11), faces = (1, 2, 3)(-1, -2, -5)(-3, 26, -25)(4, 5, 6)(-4, 22, -20)(-6, -19, 21)(7, 8, 9)(-7, -8, -12)(-9, 18, -17)(10, 11, 12)(-10, -27, -24)(-11, 16, -13)(13, 14, 15)(-14, 17, -16)(-15, 27, -26)(-18, 20, 19)(-21, 24, 23)(-22, 25, -23)) with vectors 1: ((x+1/2 ~ 1.9142136), 1), 2: ((-x-1/2 ~ -1.9142136), 0), 3: (0, -1), 4: ((-1/2 ~ -0.500000), (-3*x+5/2 ~ -1.7426407)), 5: (0, -1), 6: ((1/2 ~ 0.500000), (3*x-3/2 ~ 2.7426407)), 7: ((x+1/2 ~ 1.9142136), 1), 8: ((-x-1/2 ~ -1.9142136), 0), 9: (0, -1), 10: ((1/2 ~ 0.500000), (3/2 ~ 1.500000)), 11: ((-1/2 ~ -0.500000), (-1/2 ~ -0.500000)), 12: (0, -1), 13: ((-1/2 ~ -0.500000), (3*x-5/2 ~ 1.7426407)), 14: (0, (-3*x+3 ~ -1.2426407)), 15: ((1/2 ~ 0.500000), (-1/2 ~ -0.500000)), 16: (-1, (3*x-3 ~ 1.2426407)), 17: (-1, 0), 18: (-1, -1), 19: (-1, (-3*x+2 ~ -2.2426407)), 20: (0, (3*x-3 ~ 1.2426407)), 21: ((-1/2 ~ -0.500000), (1/2 ~ 0.500000)), 22: ((-1/2 ~ -0.500000), (-1/2 ~ -0.500000)), 23: ((-2*x-3/2 ~ -4.3284271), (1/2 ~ 0.500000)), 24: ((2*x+1 ~ 3.8284271), 0), 25: ((-2*x-2 ~ -4.8284271), 0), 26: ((-2*x-2 ~ -4.8284271), -1), 27: ((-2*x-3/2 ~ -4.3284271), (-3/2 ~ -1.500000))
```
for the direction `((x+1/2 ~ 1.9142136), 1)`.

**Comment by saraedum** [comment]
I looked into this specific one a bit and what intervalxt does looks correct. The collapsed triangulation looks reasonable
```
FlatTriangulationCombinatorial(vertices = (1, 10, 11, 16, -17, 14)(-1, -13, -11, -9, 17, 8, 3, -4, 5, 18)(2, -5, -7, 15)(-2, 6, -3, -12, 7, 4, -6, -15, 12, -8, -16, 13, -14, 9, -10, -18), faces = (1, 18, -10)(-1, 14, 13)(2, -18, 5)(-2, 15, -6)(3, 6, 4)(-3, 8, 12)(-4, 7, -5)(-7, -12, -15)(-8, 17, 16)(9, -11, 10)(-9, -14, -17)(11, -13, -16)) with vectors 1: ((-3*x-3 ~ -7.2426407), (-7/2 ~ -3.500000)) from 16 to -8, -1: ((2*x+5/2 ~ 5.3284271), (5/2 ~ 2.500000)) from 12 to 16, 2: (-1, (-3*x+2 ~ -2.2426407)) from 19 to -19, 3: (0, -1) from 5 to -5, -3: (0, 1), 4: ((-1/2 ~ -0.500000), (-3*x+5/2 ~ -1.7426407)), 5: (0, (3*x-3 ~ 1.2426407)) from 20 to -20, 6: ((1/2 ~ 0.500000), (3*x-3/2 ~ 2.7426407)), 7: ((-1/2 ~ -0.500000), (-1/2 ~ -0.500000)) from 22 to -22, 8: ((-2*x-2 ~ -4.8284271), -3) from 26 to -27, 9: ((3*x+7/2 ~ 7.7426407), 4) from -17 to 26, 10: ((-3*x-4 ~ -8.2426407), (-9/2 ~ -4.500000)) from 16 to -17, 11: ((-1/2 ~ -0.500000), (-1/2 ~ -0.500000)), 12: ((2*x+2 ~ 4.8284271), 2) from -27 to 25, 13: ((-2*x-9/2 ~ -7.3284271), (-3*x+1/2 ~ -3.7426407)) from 17 to 12, 14: (2, (3*x-3 ~ 1.2426407)) from 15 to 17, 15: ((-1/2 ~ -0.500000), (1/2 ~ 0.500000)) from 21 to -21, -15: ((2*x+3/2 ~ 4.3284271), (3/2 ~ 1.500000)) from -27 to 27, 16: ((x+7/2 ~ 4.9142136), (3*x-2 ~ 2.2426407)) from 15 to 8, -16: ((-2*x-4 ~ -6.8284271), (-3*x+1 ~ -3.2426407)) from 17 to 24, 17: ((-3*x-11/2 ~ -9.7426407), (-3*x-1 ~ -5.2426407)) from 26 to 15, 18: (-1, -1), collapsed half edges -1: (((-x-1/2 ~ -1.9142136), -1) from -7 to 7), 3: (((-x-1/2 ~ -1.9142136), -1) from -1 to 1), -3: (((x+1/2 ~ 1.9142136), 1) from 1 to -1), 15: (((2*x+1 ~ 3.8284271), 2) from 24 to 27), 16: (((-2*x-1 ~ -3.8284271), -2) from 27 to 24), -16: (((x+1/2 ~ 1.9142136), 1) from 7 to -7) with respect to ((x+1/2 ~ 1.9142136), 1)
```

At some point, we see this single flow component:
```
Component with perimeter [((-9*x-14 ~ -26.727922), (-6*x-11/2 ~ -13.985281)) from 21 to -17 → ((3*x+7/2 ~ 7.7426407), 4) from -17 to 26 → ((2*x+9/2 ~ 7.3284271), (3*x-1/2 ~ 3.7426407)) from 12 to 17 → ((x+1/2 ~ 1.9142136), 1) from 7 to -7 → ((-6*x-21/2 ~ -18.985281), (-6*x-3/2 ~ -9.9852814)) from 25 to 15 → ((6*x+10 ~ 18.485281), (6*x+1 ~ 9.4852814)) from 15 to 27 → (-2, (-3*x+3 ~ -1.2426407)) from 21 to 5 → ((2*x+9/2 ~ 7.3284271), (3*x-1/2 ~ 3.7426407)) from 5 to 25 → ((x+1/2 ~ 1.9142136), 1) from 1 to -1 → ((9*x+14 ~ 26.727922), (6*x+11/2 ~ 13.985281)) from -17 to 21 → ((2*x+1 ~ 3.8284271), 2) from 24 to 27 → ((-6*x-10 ~ -18.485281), (-6*x-1 ~ -9.4852814)) from 27 to 15 → ((-2*x-1 ~ -3.8284271), -2) from 27 to 24 → ((6*x+21/2 ~ 18.985281), (6*x+3/2 ~ 9.9852814)) from 15 to 25 → ((-2*x-9/2 ~ -7.3284271), (-3*x+1/2 ~ -3.7426407)) from 25 to 5 → ((-x-1/2 ~ -1.9142136), -1) from -1 to 1 → ((-3*x-7/2 ~ -7.7426407), -4) from 26 to -17 → ((-2*x-9/2 ~ -7.3284271), (-3*x+1/2 ~ -3.7426407)) from 17 to 12 → ((-x-1/2 ~ -1.9142136), -1) from -7 to 7 → (2, (3*x-3 ~ 1.2426407)) from 5 to 21]
```

But after one step of zorich  induction, things are wrong:
```
[5: (-3/2*x+5/2 ~ 0.37867966)] [13: (x-5/4 ~ 0.16421356)] [9: (-x+3/2 ~ 0.085786438)] [12: (x-5/4 ~ 0.16421356)] [17: (-3/2*x+9/4 ~ 0.12867966)] [7: (2*x-5/2 ~ 0.32842712)] [10: (-1/2*x+3/4 ~ 0.042893219)] / [10] [9] [13] [17] [7] [5] [12]
Changing Length (-2, (-3*x+3 ~ -1.2426407)) from 21 to 5
Changed Length to ((3*x+7/2 ~ 7.7426407), 4) from 12 to 20
[5: (-x+3/2 ~ 0.085786438)] [13: (x-5/4 ~ 0.16421356)] [9: (-x+3/2 ~ 0.085786438)] [12: (x-5/4 ~ 0.16421356)] [17: (-3/2*x+9/4 ~ 0.12867966)] [7: (2*x-5/2 ~ 0.32842712)] [10: (-1/2*x+3/4 ~ 0.042893219)] / [17] [7] [10] [9] [13] [5] [12]
Changing Length ((-6*x-21/2 ~ -18.985281), (-6*x-3/2 ~ -9.9852814)) from 25 to 15
Changed Length to ((-9*x-14 ~ -26.727922), (-6*x-11/2 ~ -13.985281)) from 20 to 15
[13: (x-5/4 ~ 0.16421356)] [9: (-x+3/2 ~ 0.085786438)] [12: (x-5/4 ~ 0.16421356)] [5: (-x+3/2 ~ 0.085786438)] [17: (-1/2*x+3/4 ~ 0.042893219)] [7: (2*x-5/2 ~ 0.32842712)] [10: (-1/2*x+3/4 ~ 0.042893219)] / [17] [7] [10] [9] [13] [5] [12]
```

Afterwards, the error occurs:
```
segment + 1 == end(path) || connected(*segment, *(segment + 1)) does not hold: Path must be connected but ((x+1/2 ~ 1.9142136), 1) from 7 to -7 does not precede ((3*x+7/2 ~ 7.7426407), 4) from 12 to 20 either because they are connected to different vertices or because the turn from ((-x-1/2 ~ -1.9142136), -1) from -7 to 7 to ((3*x+7/2 ~ 7.7426407), 4) from 12 to 20 is not turning clockwise in the range (0, 2π] in ../../../flatsurf/libflatsurf/src/path.cc:158
```

**Comment by saraedum** [comment]
Fixed in #110
